activerecord/test/cases/relation/where_chain_test.rb

```diff
@@ -153,5 +153,33 @@ def test_rewhere_with_infinite_range
 
       assert_equal expected.to_a, relation.to_a
     end
+
+    def test_where_satisfies
+      relation = Post.where.satisfies { |post| (post.author_id != 1).and(post.id > 2) }
```

Comment:
I really like the `wharel` method, the `satisfies` after the `where` seems confusing (at least to me). Do Rails provide another API similar to this one?